publish_courses/static/publish_courses/js/controllers/IndexController.js

```diff
@@ -83,5 +89,7 @@
             $scope.clearMessages();
             $scope.loadCoursesSummary();
         };
+
+        $scope.initialize();
```

Comment:
No, it was just happening in the main body of the controller (when the controller was instantiated). By moving it into an initialize() function, the order of the `$scope` function declarations doesn't matter anymore (otherwise any functions called by the initialization code would need to be defined above that code in the main body of the controller function). (Incidentally, `ng-init` is [not recommended for kicking off controller code](https://docs.angularjs.org/api/ng/directive/ngInit), apparently -- I was interested in that too, because I thought then jasmine tests would be a little easier.)